apps/token/src/routes/rewards/epoch-total-rewards/epoch-rewards.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import { AsyncRenderer } from '@vegaprotocol/ui-toolkit';
+import { useEpochAssetsRewardsQuery } from '../home/__generated__/Rewards';
+import { useRefreshAfterEpoch } from '../../../hooks/use-refresh-after-epoch';
+import { generateEpochTotalRewardsList } from './generate-epoch-total-rewards-list';
+import { NoRewards } from '../no-rewards';
+import { EpochTotalRewardsTable } from './epoch-total-rewards-table';
+import type { AggregatedEpochSummary } from './generate-epoch-total-rewards-list';
+
+export const EpochRewards = () => {
+  const [epochRewardSummaries, setEpochRewardSummaries] = useState<
+    AggregatedEpochSummary[]
+  >([]);
+
+  const { data, loading, error, refetch } = useEpochAssetsRewardsQuery({
+    variables: {
+      epochRewardSummariesPagination: {
+        first: 10,
+      },
+    },
+  });
+  useRefreshAfterEpoch(data?.epoch.timestamps.expiry, refetch);
+
+  useEffect(() => {
```

Comment:
suggestion (non-blocking): I don't think you need the combination of `useEffect` and `useState` here.

Simple `const epochRewardSummaries = generateEpochTotalRewardsList(data) || [];` would do the same job here

Reply:
Great point! Thanks